src/coreclr/vm/interpframeallocator.cpp

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifdef FEATURE_INTERPRETER
+
+#include "interpexec.h"
+#include "interpframeallocator.h"
+
+FrameDataFragment::FrameDataFragment(size_t size)
+{
+    if (size < INTERP_STACK_FRAGMENT_SIZE)
+    {
+        size = INTERP_STACK_FRAGMENT_SIZE;
+    }
+
+    start = (uint8_t*)malloc(size);
+    assert(start && "Failed to allocate FrameDataFragment");
```

Comment:
OutOfMemory handling should not be done by asserts. It should throw exception; or if exception is not viable, fail with fail fast (`EEPOLICY_HANDLE_FATAL_ERROR`).

Reply:
Now it calls `ThrowOutOfMemory`, similar other allocations in `coreclr/vm` files. @BrzVlad is any cleanup required when this exception is thrown?

Also, does CoreCLR already implement a struct similar to FrameDataAllocator that we could reuse instead?

Reply:
In my opinion, leaving the assert and adding a FIXME like in other places where we need to throw would be the right approach for now. EH support is still in the works and some implementation details need to be figured out before, so we use the same pattern everywhere. I don't think the exception that we throw here has correct behavior right now anyway since it can't be caught by executing interpreter frames.

Reply:
Ok, reverted with a TODO comment.